### PR TITLE
Feature/red 235 add system score to grades table

### PR DESCRIPTION
**Pull request by PanopticaRising**

### CODE REVIEW

**Comment by PanopticaRising** [comment]
image
image
image

**Review by red-tt-bear — APPROVED** [review]

File: src/features/courses/course-controller.ts [hunk 1]
```diff
@@ -2381,6 +2382,7 @@ class CourseController {
                 [sequelize.fn('avg', sequelize.col(`topics.questions.grades.${StudentGrade.rawAttributes.numAttempts.field}`)), 'averageAttemptedCount'],
                 [averageScoreAttribute, 'averageScore'],
                 [sequelize.fn('count', sequelize.col(`topics.questions.grades.${StudentGrade.rawAttributes.id.field}`)), 'totalGrades'],
+                [sequelize.fn('avg', sequelize.col(`topics.questions.grades.${StudentGrade.rawAttributes.overallBestScore.field}`)), 'systemScore'],
```
Comment (doombot-exe) [inline src/features/courses/course-controller.ts:2385]: overallBestScore is not what we want -- the 'system score' should come from the partial credit best score...

File: src/features/courses/course-controller.ts [hunk 2]
```diff
@@ -2497,6 +2499,7 @@ class CourseController {
                 [sequelize.fn('avg', sequelize.col(`questions.grades.${StudentGrade.rawAttributes.numAttempts.field}`)), 'averageAttemptedCount'],
                 [averageScoreAttribute, 'averageScore'],
                 [sequelize.fn('count', sequelize.col(`questions.grades.${StudentGrade.rawAttributes.id.field}`)), 'totalGrades'],
+                [sequelize.fn('avg', sequelize.col(`questions.grades.${StudentGrade.rawAttributes.overallBestScore.field}`)), 'systemScore'],
```
Comment (doombot-exe) [inline src/features/courses/course-controller.ts:2502]: same here?

**Review by doombot-exe — CHANGES_REQUESTED** [review]

File: src/features/courses/course-controller.ts (hunk 1 above)
Comment (PanopticaRising) [reply src/features/courses/course-controller.ts:2385]: Good catches, I thought I fixed those.

**Review by doombot-exe — APPROVED** [review]

**Pull request merged**